src/components/navigationConfig.ts

```diff
@@ -133,49 +134,56 @@ export const menus = () => {
   return lastMenus
 }
 
-const saveNavigation = (i: number, k: string) => {
+const saveNavigation = (index: number, keyVal: string) => {
   const arr = menus()
-  const m = arr[i]
-  const p = getAuthStore().currentProfile
-  if (!m || !p) {
+  let menu = arr[index]
+  const { subMenusMap, defaultSubMenuKey, key } = menu
+  const { currentProfile } = getAuthStore() || {}
+  let { navIndex, navSubMenus } = currentProfile || {}
```

Comment:
These can be destructured in single line like const { currentProfile:  { navIndex, navSubMenus }} = getAuthStore() || {} and navIndex and navSubMenus can be used directly.